### ISSUE TITLE
Logout is not working in Edge browser. In Chrome it is working fine

### ISSUE THREAD

**Issue by VijayPatel1** [IdentityServer/IdentityServer4#2133]
I am using Identityserver4 and oidc-client.js for for angular4 application signout. The Identityserver4 is configured for login using external IDP using WS-Federation protocol. 
So the flow is like as below:
Identityserver 4 is configured to use WS-Federation to use ADFS as a RP and the ADFS redirect the request to other client IDP.

My client application >> Identityserver4 >> ADFS >> Other IDP

So here login/logout both working fine in Chrome.
Login working fine is EDGe browser, Logout shows successfully logged out in Edge browser. But when I type client application user in seperation tab then it automatically logged in to application. So seems to be not correctly logged out.

My configurations is as per below:

My strtup.cs file:

public void ConfigureServices(IServiceCollection services)
		{
			services.AddIdentityServer()
			.AddSigningCredential(LoadCertificate())
			.AddInMemoryApiResources(new Config(configuration, Environment).GetApiResources())
			.AddInMemoryIdentityResources(Config.GetIdentityResources())
			.AddInMemoryClients(new Config(configuration, Environment).GetClients())
			.AddAuthUserStore();

			services.AddAuthentication()
			.AddWsFederation("WsFed", options =>
			{
				ConfigureADFSWSFederationOprtions(options);
			});
		}

		private void ConfigureADFSWSFederationOprtions(WsFederationOptions options)
		{
			options.SignInScheme = IdentityServerConstants.ExternalCookieAuthenticationScheme;
			//options.SignOutScheme = IdentityServerConstants.SignoutScheme;
			options.RequireHttpsMetadata = false;
			options.Wtrealm = configuration["WSFederationRealm"];
			options.CallbackPath = "/";
			options.SkipUnrecognizedRequests = true;

			var config = new WsFederationConfiguration
			{
				Issuer = configuration["WSFederationIssuer"],
				TokenEndpoint = configuration["WSFederationTokenEndpoint"],
			};

			string certpath = string.Format("{0}\\Certificates\\{1}", Directory.GetCurrentDirectory(), configuration["WSFederationCertName"]);
			var cert = new X509Certificate2(certpath);
			var key = new X509SecurityKey(cert);

			config.SigningKeys.Add(key);
			options.Configuration = config;
			options.TokenValidationParameters = new TokenValidationParameters
			{
				ValidAudience = options.Wtrealm,
			};
			options.Events.OnTicketReceived += OnTicketReceived;

		}

public void Configure(IApplicationBuilder app, IHostingEnvironment env, ILoggerFactory loggerFactory, IAuthUserRepository authRepository, IServiceProvider services)
		{
			app.UseCors(m => m.AllowAnyOrigin().AllowAnyHeader().AllowAnyMethod());

			app.UseStaticFiles();
			app.UseIdentityServer();
			app.UseAuthentication();
			app.UseMvcWithDefaultRoute();
		}

>> Client Configuration Config.cs
new Client
                {
                    ClientId = "AngularClientExternalLogin",
                    ClientName = "Angular JS Client",
                    AllowedGrantTypes = GrantTypes.Implicit,
                    AllowAccessTokensViaBrowser = true,

                    RedirectUris = { string.Format("{0}/auth.html", erp2TaxUri), string.Format("{0}/silent-renew.html", erp2TaxUri) },
                    PostLogoutRedirectUris = { erp2TaxUri },
                    AllowedCorsOrigins = erp2TaxClientAllowdCORSOrigins,

                    AllowedScopes =
                    {
                        IdentityServerConstants.StandardScopes.OpenId,
                        IdentityServerConstants.StandardScopes.Profile,
                        "Erp2TaxService"
                    },

                    RequireConsent = false,
                    ClientSecrets =
                    {
                        new Secret(secretKey.Sha256())
                    },
                    AccessTokenLifetime = 900,
                    IdentityTokenLifetime = 300,
                    UpdateAccessTokenClaimsOnRefresh = true,
                    AccessTokenType = AccessTokenType.Reference,
                    EnableLocalLogin = false,
                    IdentityProviderRestrictions = { "WsFed" }
                },

>> Logout methods of AccountController.cs

[HttpGet]
		public async Task<IActionResult> Logout(string logoutId)
		{
			// build a model so the logout page knows what to display
			var vm = await BuildLogoutViewModelAsync(logoutId);

			if (vm.ShowLogoutPrompt == false)
			{
				// if the request for logout was properly authenticated from IdentityServer, then
				// we don't need to show the prompt and can just log the user out directly.
				return await Logout(vm);
			}

			return View(vm);
		}

		/// <summary>
		/// Handle logout page postback
		/// </summary>
		[HttpPost]
		[ValidateAntiForgeryToken]
		public async Task<IActionResult> Logout(LogoutInputModel model)
		{
			// build a model so the logged out page knows what to display
			var vm = await BuildLoggedOutViewModelAsync(model.LogoutId);

			if (User?.Identity.IsAuthenticated == true)
			{
				// delete local authentication cookie
				await HttpContext.SignOutAsync();

				_logger.LogInformation($"vm.ExternalAuthenticationScheme: {vm.ExternalAuthenticationScheme}");
				await HttpContext.SignOutAsync(vm.ExternalAuthenticationScheme);

				_logger.LogInformation($"User.GetSubjectId(): {User.GetSubjectId()}");
				_logger.LogInformation($"User.GetDisplayName(): {User.GetDisplayName()}");

				// raise the logout event
				await _events.RaiseAsync(new UserLogoutSuccessEvent(User.GetSubjectId(), User.GetDisplayName()));
			}

			// check if we need to trigger sign-out at an upstream identity provider
			if (vm.TriggerExternalSignout)
			{
				// build a return URL so the upstream provider will redirect back
				// to us after the user has logged out. this allows us to then
				// complete our single sign-out processing.
				string url = Url.Action("Logout", new { logoutId = vm.LogoutId });

				// this triggers a redirect to the external provider for sign-out
				return SignOut(new AuthenticationProperties { RedirectUri = url }, vm.ExternalAuthenticationScheme);
			}

			return View("LoggedOut", vm);
		}

Can you please suggest here is there something issue from my configuration side?

**Comment by hadiidbouk** [comment]
It happens to me also after login

**Comment by brockallen** [comment]
I just tested this locally with edge with my client => IdentityServer => ADFS4 using WS-Fed. Signout is working properly for me. I don't have the additional upstream IdP from my ADFS4, so I don't know if that's your issue. 

In short, this is very hard for me to diagnose on the issue tracker. I think you will need to dig more into yourself. I'd suggest first doing it all in chrome and fully understand where all the connections are being made to make it successful, then go back into edge and find out why in that scenario which connection is not working.

Very commonly, due to the "Zones" feature in Windows' Internet Options, Edge and IE will refuse to pass cookies when opening iframes back to various sites. Unfortunately many of the single sign-out protocols rely upon this capability. 

Let us know what you find.

**Comment by chrisowhite** [comment]
@VijayPatel1 I think @brockallen might be onto something with Windows internet options, particularly protected mode. We've had a few QA analysts at work have trouble testing in Edge because for some reason group policy enforces certain settings in Edge/IE.

**Comment by brockallen** [comment]
Closing. If you still have issues, feel free to reopen.

**Comment by smsaxena5** [comment]
I am having the same problem with the Edge broswer. I am able to log out of my Asp.Net core MVC application and I am also automatically being redirected to the IdentityServer login page. I see the following message on the console:  [Information] Showing login: User is not authenticated.
If I then try to go to a url of my MVC app from which I logged out I am being redirected to the IdentityServer login page on the first try. However on the second try I am able to open it as an authenticated user.

Here is something else I noticed: The first time when I am not allowed to open the url, the OpenIdConnect OnRedirectToIdentityProvider event handler in my mvc app is being called. The second time around it is not being called.
